src/dmd/root/response.d

```diff
@@ -14,200 +14,341 @@
 
 module dmd.root.response;
 
-import core.stdc.stdio;
-import core.stdc.stdlib;
-import core.stdc.string;
 import dmd.root.file;
 import dmd.root.filename;
 
+alias responseExpand = responseExpandFrom!lookupInEnvironment;
+
 /*********************************
- * #include <stdlib.h>
- * int response_expand(int *pargc,char ***pargv);
- *
  * Expand any response files in command line.
  * Response files are arguments that look like:
  *   @NAME
- * The name is first searched for in the environment. If it is not
- * there, it is searched for as a file name.
  * Arguments are separated by spaces, tabs, or newlines. These can be
- * imbedded within arguments by enclosing the argument in '' or "".
+ * imbedded within arguments by enclosing the argument in "".
+ * Backslashes can be used to escape a "
+ * A line comment can be started with #.
  * Recursively expands nested response files.
  *
- * To use, put the line:
- *   response_expand(&argc,&argv);
- * as the first executable statement in main(int argc, char **argv).
- * argc and argv are adjusted to be the new command line arguments
- * after response file expansion.
+ * To use, put the arguments in a Strings object and call this on it.
  *
  * Digital Mars's MAKE program can be notified that a program can accept
  * long command lines via environment variables by preceding the rule
  * line for the program with a *.
  *
+ * Params:
+ *     lookup = alias to a function that is called to look up response file
+ *              arguments in the environment. It is expected to accept a null-
+ *              terminated string and return a mutable char[] that ends with
+ *              a null-terminator or null if the response file could not be
+ *              resolved.
+ *     args = array containing arguments as null-terminated strings
+ *
  * Returns:
- *   0   success
- *   !=0   failure (argc, argv unchanged)
+ *     false on success, true if a response file could not be expanded.
```

Comment:
`0` for success makes sense for a C function returning `int`. `false` on success for a D function is heresy (we do commit heresy in DMD, but it's not something we should promote).

Reply:
I tried to not make any behavioural changes, but I can invert it.